packages/protocol/contracts/L1/verifiers/SgxVerifier.sol

```diff
@@ -32,10 +34,14 @@ contract SgxVerifier is EssentialContract, IVerifier {
     /// bootstrapping the network with trustworthy instances.
     struct Instance {
         address addr;
-        uint64 addedAt; // We can calculate if expired
+        uint64 addedAt; // We can calculate if expired or not yet valid
+        bool checkValidityCoolTime; // Owner added sgx instances not needed to have a cooldown for
+            // validity
```

Comment:
In raiko, should we make this expire-register rotation happen automatically? 
Pro is all auto, raiko can read the INSTANCE_EXPIRY setting and rotate before that.
Con is prover needs to config their private key into raiko to call register, and we need to handle the error cases which make raiko more complex.
Maybe keep raiko simpler is better.

BTW: now the register is done manually, through that forge script config_dcap_xx.s.sol.

Reply:
> (I was also thinking first to have a `validUntil` instead of doing `validFrom + INSTANCE_EXPIRY`, but doing the `validFrom + INSTANCE_EXPIRY` seems more flexible so we can update `INSTANCE_EXPIRY` at any time by updating the smart contract and then all instances automatically have the updated lifetime which seems more convenient to me)

Good one! Added.